sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -741,16 +762,20 @@ object HiveExternalCatalog {
   val STATISTICS_NUM_ROWS = STATISTICS_PREFIX + "numRows"
   val STATISTICS_COL_STATS_PREFIX = STATISTICS_PREFIX + "colStats."
 
-  def removeStatsProperties(metadata: CatalogTable): Map[String, String] = {
-    metadata.properties.filterNot { case (key, _) => key.startsWith(STATISTICS_PREFIX) }
+  // Ideally we should use `spark.sql.sources.location` to store the table location, but as we have
+  // already used `path` to store it, we should keep it for backward compatibility.
+  val TABLE_LOCATION = "path"
```

Comment:
We still have the "path" literal hard-coded in several places in different files. Should this be pulled out into a common constant somewhere?